Actions/FetchProjectDependencies/action.yaml

```diff
@@ -0,0 +1,45 @@
+name: Fetch Project dependencies
+author: Microsoft Corporation
+description: Fetches the dependencies of a project
+inputs:
+  shell:
+    description: Shell in which you want to run the action (powershell or pwsh)
+    required: false
+    default: powershell
+  project:
+    description: Project for which to fetch dependencies
+    required: true
+  buildMode:
+    description: Build mode used when building the artifacts
+    required: true
+  projectsDependenciesJson:
+    description: A JSON object that matches eash project with its dependency projects
+    required: true
+  destinationPath:
+    default: ${{ github.workspace }}/.dependencies
+    description: The destination path where the dependencies will be stored
+outputs:
+  FetchedApps: 
+    description: A JSON-formatted array of paths to .app files of the apps that were fetched
+    value: ${{ steps.FetchDependencies.outputs.FetchedArtifacts }}
+  FetchedTestApps: 
+    description: A JSON-formatted array of paths to .app files of the test apps that were fetched
+    value: ${{ steps.FetchDependencies.outputs.FetchedArtifacts }}
+runs:
+  using: composite
+  steps:
+    - name: Download artifacts from current build
+      uses: actions/download-artifact@v3
+      with:
+        path: ${{ inputs.destinationPath}}
+
+    - name: Fetch project dependencies
+      shell: ${{ inputs.shell }}
+      id: FetchDependencies
+      env:
+        gitHubToken: ${{ github.token }}
+      run:
+        ${{ github.action_path }}/FetchProjectDependencies.ps1 -project '${{ inputs.project }}' -buildMode '${{ inputs.buildMode }}' -projectsDependenciesJson '${{ inputs.projectsDependenciesJson }}' -baseBranch '${{ github.base_ref }}' -baseFolder '${{ github.workspace }}'' -destinationPath '${{ inputs.destinationPath }}' -token '${{ github.token }}'
```

Comment:
inputs should never be used directly in code like this.
inputs should be transferred to env vars and used as env for correct UTF8 conversion, but also for security reasons.

Reply:
Isn't it an issue only when using inline scripts directly in the workflow?

> [Using an action instead of an inline script (recommended)](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-an-action-instead-of-an-inline-script-recommended)
The recommended approach is to create an action that processes the context value as an argument. This approach is not vulnerable to the injection attack, as the context value is not used to generate a shell script, but is instead passed to the action as an argument

Reply:
That might be, but you will have problems if the input contains UFT8 characters, " or ' - I think we should always use the same pattern as the other actions.

Reply:
Switched to using env variables.